app/src/main/java/com/gophillygo/app/utils/GpgLocationUtils.java

```diff
@@ -0,0 +1,122 @@
+package com.gophillygo.app.utils;
+
+import android.Manifest;
+import android.app.Activity;
+import android.app.Dialog;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.location.Location;
+import android.support.v4.app.ActivityCompat;
+import android.util.Log;
+import android.widget.Toast;
+
+import com.google.android.gms.common.ConnectionResult;
+import com.google.android.gms.common.GoogleApiAvailability;
+import com.google.android.gms.location.FusedLocationProviderClient;
+import com.google.android.gms.location.LocationRequest;
+import com.google.android.gms.location.LocationServices;
+import com.gophillygo.app.R;
+
+import java.lang.ref.WeakReference;
+
+
+public class GpgLocationUtils {
+
+    public interface LocationUpdateListener {
+        void locationFound(Location location);
+    }
+
+    // identifier for device location access request, if runtime prompt necessary
+    // request code must be in lower 8 bits
+    public static final int PERMISSION_REQUEST_ID = 11;
+    private static final int API_AVAILABILITY_REQUEST_ID = 22;
+
+    private static final int LOCATION_REQUESTS_COUNT = 12;
```

Comment:
Is there a reason we request 12 updates? I would have assumed we could get away with only 1.

Reply:
Not sure if it was something to do with the emulator, but sometimes a single request would return nothing. My reasoning is that at least this shouldn't hurt, and might give the GPS time to settle slightly if it had not been fired up in a while.